subprojects/gst-plugins-good/sys/v4l2/gstv4l2allocator.c

```diff
@@ -1023,6 +1029,97 @@ gst_v4l2_allocator_alloc_dmabufin (GstV4l2Allocator * allocator)
   return group;
 }
 
+GstV4l2MemoryGroup *
+gst_v4l2_allocator_alloc_dmabufin_capture (GstV4l2Allocator * allocator,
+    GstAllocator * dmabuf_allocator, GstBuffer* downstream_buffer)
+{
+#if 1
+  GstV4l2Object *obj = allocator->obj;
+  GstV4l2MemoryGroup *group;
+  gint i;
+
+  g_return_val_if_fail (allocator->memory == V4L2_MEMORY_DMABUF, NULL);
+
+  group = gst_v4l2_allocator_alloc (allocator);
+
+  if (group == NULL)
+    return NULL;
+
+  for (i = 0; i < group->n_mem; i++) {
+    GstV4l2Memory *mem;
+    GstMemory *dma_mem;
+    gsize size, offset, maxsize;
+
+
+    if (group->mem[i] == NULL) {
+      dma_mem = gst_buffer_peek_memory (downstream_buffer, i);
+      int downstream_fd = gst_dmabuf_memory_get_fd (dma_mem);
+      int dup_fd = dup (downstream_fd);
```

Comment:
This will be leaked as we no longer close dmabuf FD in dmabuf import mode. You also ref the foreign memory and store in as a quark, so you really don't have to dup the FD, you can borrow it instead.